plugins/redis/redis.cpp

```diff
@@ -515,38 +525,68 @@ void Connection::assertConnection(const char * _msg)
     //NOTE: documented as never failing, but in case
     assertOnErrorWithCmdMsg(reply->query(), "FlushDB");
 }
 void Connection::del(ICodeContext * ctx, const char * key)
 {
-    OwnedReply reply = Reply::createReply(redisCommand(context, "DEL %b", key, strlen(key)));
+    OwnedReply reply = Reply::createReply(redisCommand("DEL %b", key, strlen(key)));
     assertOnErrorWithCmdMsg(reply->query(), "Del", key);
 }
 void Connection::persist(ICodeContext * ctx, const char * key)
 {
-    OwnedReply reply = Reply::createReply(redisCommand(context, "PERSIST %b", key, strlen(key)));
+    OwnedReply reply = Reply::createReply(redisCommand("PERSIST %b", key, strlen(key)));
     assertOnErrorWithCmdMsg(reply->query(), "Persist", key);
 }
 void Connection::expire(ICodeContext * ctx, const char * key, unsigned _expire)
 {
-    OwnedReply reply = Reply::createReply(redisCommand(context, "PEXPIRE %b %u", key, strlen(key), _expire));
+    OwnedReply reply = Reply::createReply(redisCommand("PEXPIRE %b %u", key, strlen(key), _expire));
     assertOnErrorWithCmdMsg(reply->query(), "Expire", key);
 }
 bool Connection::exists(ICodeContext * ctx, const char * key)
 {
-    OwnedReply reply = Reply::createReply(redisCommand(context, "EXISTS %b", key, strlen(key)));
+    OwnedReply reply = Reply::createReply(redisCommand("EXISTS %b", key, strlen(key)));
     assertOnErrorWithCmdMsg(reply->query(), "Exists", key);
     return (reply->query()->integer != 0);
 }
 unsigned __int64 Connection::dbSize(ICodeContext * ctx)
 {
-    OwnedReply reply = Reply::createReply(redisCommand(context, "DBSIZE"));
+    OwnedReply reply = Reply::createReply(redisCommand("DBSIZE"));
     assertOnErrorWithCmdMsg(reply->query(), "DBSIZE");
     return reply->query()->integer;
 }
+signed __int64 Connection::incrBy(ICodeContext * ctx, const char * key, signed __int64 value)
+{
+    OwnedReply reply = Reply::createReply(redisCommand("INCRBY %b %" I64F "d", key, strlen(key), value));
+    return returnInt(key, "INCRBY", reply->query());
+}
 //-------------------------------------------SET-----------------------------------------
+void Connection::setInt(ICodeContext * ctx, const char * key, signed __int64 value, unsigned expire, bool _unsigned)
+{
+    StringBuffer cmd("SET %b %" I64F);
+    if (_unsigned)
+        cmd.append("u");
+    else
+        cmd.append("d");
+
+    appendExpire(cmd, expire);
+
+    OwnedReply reply = Reply::createReply(redisCommand(cmd.str(), key, strlen(key), value));
+    assertOnErrorWithCmdMsg(reply->query(), "SET", key);
+}
+void Connection::setReal(ICodeContext * ctx, const char * key, double value, unsigned expire)
+{
+    /*MORE: I really think that this should be 15 and that we are printing 1 to many (i.e. 16) significant digits for results.
+     * This may be related to https://track.hpccsystems.com/browse/HPCC-12177 "Instability in floating point calculations for statistical functions".
```

Comment:
I don't think this comment is relevant. It isn't related to instability in numeric calculations. And that issue is also not related to storing results as text.